packages/celotool/src/lib/generate_utils.ts

```diff
@@ -433,3 +436,81 @@ export const generateGenesis = ({
 
   return JSON.stringify(genesis, null, 2)
 }
+
+// This function assumes that mycelo has already been built using 'make all'
+export const generateGenesisWithMigrations = async ({
+  gethRepoPath,
+  genesisConfig,
+  mnemonic,
+  numValidators,
+}: MyceloGenesisConfig): Promise<string> => {
+  const tmpDir = path.join(tmpdir(), `mycelo-genesis-${Date.now()}`)
+  fs.mkdirSync(tmpDir)
+  const envFile = path.join(tmpDir, 'env.json')
+  const configFile = path.join(tmpDir, 'genesis-config.json')
+  const myceloBinaryPath = path.join(gethRepoPath!, '/build/bin/mycelo')
+  await spawnCmdWithExitOnFailure(
+    myceloBinaryPath,
+    ['genesis-config', '--template', 'monorepo', '--mnemonic', mnemonic],
+    {
+      silent: true,
+      cwd: tmpDir,
+    }
+  )
+  const mcEnv = JSON.parse(fs.readFileSync(envFile).toString())
+  const mcConfig = JSON.parse(fs.readFileSync(configFile).toString())
+
+  // Customize and overwrite the env.json file
+  merge(mcEnv, {
+    chainId: genesisConfig.chainId,
+    accounts: {
+      validators: numValidators,
+    },
+  })
+  fs.writeFileSync(envFile, JSON.stringify(mcEnv, undefined, 2))
+
+  // Customize and overwrite the genesis-config.json file
+  if (genesisConfig.chainId) {
+    mcConfig.chainId = genesisConfig.chainId
+  }
+  if (genesisConfig.epoch) {
+    mcConfig.istanbul.epoch = genesisConfig.epoch
+  }
+  if (genesisConfig.lookbackwindow) {
+    mcConfig.istanbul.lookbackwindow = genesisConfig.lookbackwindow
+  }
+  if (genesisConfig.blockTime) {
+    mcConfig.istanbul.blockperiod = genesisConfig.blockTime
+  }
+  if (genesisConfig.requestTimeout) {
+    mcConfig.istanbul.requesttimeout = genesisConfig.requestTimeout
+  }
+  if (genesisConfig.churritoBlock !== undefined) {
+    mcConfig.hardforks.churritoBlock = genesisConfig.churritoBlock
+  }
+  if (genesisConfig.donutBlock !== undefined) {
+    mcConfig.hardforks.donutBlock = genesisConfig.donutBlock
+  }
+  if (genesisConfig.timestamp !== undefined) {
+    mcConfig.genesisTimestamp = genesisConfig.timestamp
+  }
+
+  // TODO: overrides for migrations
```

Comment:
Is this a todo for when other tests are updated?

Reply:
It is required to make the functionality complete, and I believe it is in fact required for some of the other tests.  So, yes.  I will come back to it once I'm done writing tests for CIP-35.